sensu/plugins/check-neutron-agents-duplicate.py

```diff
@@ -0,0 +1,134 @@
+        print("WARN: \"%s\" returned %s: %s" % (command, p.returncode, stderr))
+        sys.exit(STATE_WARNING)
+
+def request_token(url, method='GET', retries=2, **kwargs):
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code :
+                break
+    except requests.exceptions.RequestException as e:
+        print("%s returned %s" % (url, e))
+
+    return r
+
+def request(url, method='GET', retries=2, **kwargs):
+
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code:
+                break
+    except requests.exceptions.RequestException as e:
+        print("%s returned %s" % (url, e))
+
+    return r.json()
+
+
+def check_agents(agent_list):
+
+    agent_dictionary = {}
+    
+    for agent in agent_list['agents']:
+        agent_type = agent['agent_type']
+        host = agent['host'].split('.')[0]
+
+        if agent_type  in agent_dictionary :
+            if host in agent_dictionary[agent_type] :
+                print("Duplicate agent: %s on host: %s" % (agent_type, host))
+                sys.exit(STATE_CRITICAL)
+            else:
+                agent_dictionary[agent_type].add(host)
+        else:
+            agent_dictionary[agent_type] = set()
+            agent_dictionary[agent_type].add(host)
+
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-u', '--user', default=os.environ['OS_USERNAME'])
+    parser.add_argument('-p', '--password', default=os.environ['OS_PASSWORD'])
+    parser.add_argument('-t', '--tenant', default=os.environ['OS_TENANT_NAME'])
+    parser.add_argument('-a', '--auth-url', default=os.environ['OS_AUTH_URL'])
+    args = parser.parse_args()
+
+    url = args.auth_url + '/tokens'
+    headers = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+    data = json.dumps(
+        {
+            'auth': {
+                'tenantName': args.tenant,
+                'passwordCredentials': {
+                    'username': args.user,
+                    'password': args.password
+                }
+            }
+        })
+
+    r = request_token(url, 'POST', 4, data=data, headers=headers)
+    token = None
+    if r:
+        access = r.json()['access']
+        token = access['token']['id']
+
+    if not token:
+        sys.exit(STATE_CRITICAL)
+
+    endpoints = {}
+    for service in access['serviceCatalog']:
+        for endpoint in service['endpoints']:
+            endpoints[service['name']] = endpoint.get('internalURL')
+
+    headers = {'Accept': 'application/json', 'X-Auth-Project-Id': args.tenant,
+               'X-Auth-Token': token}
+
+    endpoint = endpoints['neutron'] + '/v2.0/agents.json'
+    agent_list = request(endpoint, headers=headers )
```

Comment:
If the neutron agents API fails for any reason, I think we would get back an empty result here, and check_agents() would pass. So might be good to check the returned agent_list is not empty and warn if it is.